### PR TITLE
firewall: Add security-zones.rst explaining zone based firewall rules

### PR DESCRIPTION
**Pull request by Monviech**
Fixes: https://github.com/opnsense/docs/issues/606

Requires:
https://github.com/opnsense/core/commit/4789c2a7527190d3848510a1af7fccb90761eb7d
https://github.com/opnsense/core/commit/1293c51187768ca527ec337771a35e8ea16afcc9
https://github.com/opnsense/core/commit/0dac1d62012dac4ce0f8b00754586cf381caee9b

### CODE REVIEW

File: source/manual/firewall_groups.rst [hunk 1]
```diff
@@ -5,21 +5,21 @@
...
+    For multiwan setups be careful with groups. Since groups are not bound to a specific interface, they will
     use the normal routing system to determine the next hop when applied on WAN type interfaces (:code:`reply-to` is not used here).
 
 
 --------------------
 Settings
 --------------------
 
-Groups are actually pretty easy to setup and don't require a lot of settings.
+Groups are simple to setup and do not require a lot of settings.
```
Comment (fichtner) [inline source/manual/firewall_groups.rst:22]: typically such filler sentences are better removed as they give more clarity...

File: source/manual/how-tos/security-zones.rst [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+.. contents::
+   :local:
+   :depth: 2
+
+
+----------------------------
+Introduction
+----------------------------
+
+Firewalls manage traffic between network segments. Utilizing **zones** simplifies configurations by grouping interfaces with similar security trust levels. This approach is especially beneficial when managing numerous interfaces that require a consistent and unified ruleset.
```
Comment (fichtner) [inline source/manual/how-tos/security-zones.rst:18]:
```suggestion
Firewalls manage traffic between network segments. Utilizing **zones** simplifies configurations by grouping interfaces with similar security trust levels. This approach is beneficial when managing numerous interfaces that require a consistent and unified ruleset.
```

opinion ;)

File: source/manual/how-tos/security-zones.rst [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+When a packet arrives at a firewall interface, it is assigned to a zone based on the inbound interface. This **packet tagging** process involves:
+
+1. Determining the receiving interface
+2. Assigning the packet to the interface's configured zone
+3. Applying security policies between the source and destination zones
+
+======================  ================
+Zone (Interface Group)  Interfaces
+======================  ================
+**Trust**               eth0, eth1, eth2
```
Comment (fichtner) [inline source/manual/how-tos/security-zones.rst:41]: I'm split on using eth0 which is sort of linux nomenclature. Also not sure if you mean _interfaces_ or _devices_.

File: source/manual/how-tos/security-zones.rst [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+will be created in :menuselection:`Firewall --> Rules --> Floating` to match first.
+
+The zones from the introduction will be used. Our goal is the following ruleset:
+
+- Rule 1: Allowing `Trust`, `Untrust` and `Wifi` to use ICMP between their zones
+- Rule 2: Allowing `Trust` to access all zones, including itself, with no restrictions
+- Rule 3: Allowing `Wifi` to access `Untrust`
+- Rule 4: Allowing `Untrust` to access HTTPS in `Trust`
+- Rule 5: Allowing `vlan0.30` to access HTTPS in `vlan0.39`
+- Rule 6: Allowing `vlan0.25` to access SSH to `172.16.1.10`
```
Comment (fichtner) [inline source/manual/how-tos/security-zones.rst:95]: this appears out of scope of zoning -- a table at the beginning with zones (grouns) interfaces and devices would probably help clear things up.

File: source/manual/how-tos/security-zones.rst [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+We have multiple VLANs that will be unified into security trust zones:
+
+- `vlan0.10` to `vlan0.19` contain trusted internal networks
+- `vlan0.20` to `vlan0.29` contain untrusted external networks
+- `vlan0.30` to `vlan0.39` contain wireless networks
+
+
+Setup Interface Groups
```
Comment (fichtner) [inline source/manual/how-tos/security-zones.rst:104]: this seems like top down but bottom up would be better: devices (and interface) and then how they belong to zones (groups)

**Review by fichtner — COMMENTED** [review]
some early comments

**Comment by Monviech** [comment]
Note to myself, add link here:

https://github.com/opnsense/docs/blob/88e4dc34705a45f342113a7feb2824cbef9431bf/source/manual/how-tos/vlan_and_lagg.rst?plain=1#L214

File: source/manual/how-tos/security-zones.rst (hunk 3 above)
Comment (Monviech) [reply source/manual/how-tos/security-zones.rst:41]: Sorry missed that, gonna use igc

File: source/manual/firewall_groups.rst [hunk 6]
```diff
@@ -5,21 +5,21 @@
 To simplify rulesets, you can combine interfaces into **Interface Groups** and add policies which will be applied to
 all interfaces in the group.
 
-Since interface groups are processed before normal interfaces, you shouldn't have issues with overlapping rules in
+Since interface groups are processed before normal interfaces, you should not have issues with overlapping rules in
```
[the remark this answers is not in the document]
Comment (Monviech) [reply source/manual/firewall_groups.rst:8]: mustn't :)

File: source/manual/how-tos/security-zones.rst (hunk 4 above)
Comment (Monviech) [reply source/manual/how-tos/security-zones.rst:95]: I just need examples for intrazone and interzone. I thought what I chose here made sense.

File: source/manual/how-tos/security-zones.rst (hunk 5 above)
Comment (Monviech) [reply source/manual/how-tos/security-zones.rst:104]: Can you briefly explain the difference between interface and device since you recently worked on the terminology update.

Comment (fichtner) [reply source/manual/how-tos/security-zones.rst:104]: It's been the internal style for a long time but in a nutshell:

"device" is the OS level interface with the driver name and interface number: igb0, vlan0.12, pppoe1 -- everything you see in ifconfig.

"interface" is the assigned device with the additional interface configuration as stored in the config.xml

In most cases the device is non-functional unless assigned to an interface. There are exceptions to this rule which is lagg and bridge ;)

**Review by AdSchellevis — APPROVED** [review]
